### PR TITLE
add missing includes

### PR DESCRIPTION
**Pull request by davidlange6**
The modules IB started having compilation errors after #31704 was merged. I believe (now confirmed) this is due to a missing Eigen dependency in the buildfile of DataFormats/CaloRecoHit. I'm adding other buildfile fixes that I found along the way..

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

On Oct 24, 2020 2:48 PM, cmsbuild <notifications@github.com> wrote:

A new Pull Request was created by @davidlange6<https://github.com/davidlange6> (David Lange) for master.

It involves the following packages:

DataFormats/CaloRecHit
DataFormats/FTLRecHit
RecoTracker/TkSeedGenerator

@perrotta<https://github.com/perrotta>, @jpata<https://github.com/jpata>, @cmsbuild<https://github.com/cmsbuild>, @kpedro88<https://github.com/kpedro88>, @slava77<https://github.com/slava77> can you please review it and eventually sign? Thanks.
@fabiocos<https://github.com/fabiocos>, @makortel<https://github.com/makortel>, @felicepantaleo<https://github.com/felicepantaleo>, @GiacomoSguazzoni<https://github.com/GiacomoSguazzoni>, @JanFSchulte<https://github.com/JanFSchulte>, @rovere<https://github.com/rovere>, @VinInn<https://github.com/VinInn>, @mschrode<https://github.com/mschrode>, @gpetruc<https://github.com/gpetruc>, @ebrondol<https://github.com/ebrondol>, @mtosi<https://github.com/mtosi>, @dgulhan<https://github.com/dgulhan> this is something you requested to watch as well.
@silviodonato<https://github.com/silviodonato>, @dpiparo<https://github.com/dpiparo>, @qliphy<https://github.com/qliphy> you are the release manager for this.

cms-bot commands are listed here<http://cms-sw.github.io/cms-bot-cmssw-cmds.html>

File: RecoTracker/TkSeedGenerator/plugins/BuildFile.xml [hunk 1]
```diff
@@ -5,6 +5,42 @@
...
+<use name="RecoTracker/TkHitPairs"/>
+<use name="RecoTracker/TkSeedingLayers"/>
+<use name="RecoTracker/TransientTrackingRecHit"/>
+<use name="TrackingTools/GeomPropagators"/>
+<use name="TrackingTools/KalmanUpdators"/>
+<use name="TrackingTools/MeasurementDet"/>
+<use name="TrackingTools/PatternTools"/>
+<use name="TrackingTools/Records"/>
+<use name="TrackingTools/TrajectoryState"/>
+<use name="TrackingTools/TransientTrackingRecHit"/>
```
Comment (slava77) [inline RecoTracker/TkSeedGenerator/plugins/BuildFile.xml:43]: many of these dependencies are declared in RecoTracker/TkSeedGenerator.
Is it time to cleanup that package dependencies?

Comment (davidlange6) [reply RecoTracker/TkSeedGenerator/plugins/BuildFile.xml:43]: its correct to have dependencies in both src and plugins BuildFiles assuming both src and plugins use them directly (at least that is my understanding of the correct way to manage BuildFiles).

**Comment by kpedro88** [comment]
+upgrade

File: RecoTracker/TkSeedGenerator/plugins/BuildFile.xml [hunk 2]
```diff
@@ -5,6 +5,42 @@
...
+<use name="RecoTracker/MeasurementDet"/>
+<use name="RecoTracker/Record"/>
+<use name="RecoTracker/SpecialSeedGenerators"/>
+<use name="RecoTracker/TkHitPairs"/>
+<use name="RecoTracker/TkSeedingLayers"/>
+<use name="RecoTracker/TransientTrackingRecHit"/>
+<use name="TrackingTools/GeomPropagators"/>
+<use name="TrackingTools/KalmanUpdators"/>
+<use name="TrackingTools/MeasurementDet"/>
+<use name="TrackingTools/PatternTools"/>
```
Comment (slava77) [inline RecoTracker/TkSeedGenerator/plugins/BuildFile.xml:40]:
```suggestion
```
apparently not used in CMSSW_11_1_X_2020-10-27-1100

File: RecoTracker/TkSeedGenerator/plugins/BuildFile.xml [hunk 3]
```diff
@@ -5,6 +5,42 @@
...
 <use name="RecoPixelVertexing/PixelTrackFitting"/>
 <use name="RecoPixelVertexing/PixelLowPtUtilities"/>
+<use name="CommonTools/Utils"/>
+<use name="DataFormats/BeamSpot"/>
+<use name="DataFormats/Common"/>
+<use name="DataFormats/Math"/>
+<use name="DataFormats/SiPixelDetId"/>
+<use name="DataFormats/SiStripDetId"/>
+<use name="DataFormats/TrackReco"/>
+<use name="DataFormats/TrackerCommon"/>
```
Comment (slava77) [inline RecoTracker/TkSeedGenerator/plugins/BuildFile.xml:15]:
```suggestion
```
not used

**Review by slava77 — COMMENTED** [review]
it looks like several dependencies in  RecoTracker/TkSeedGenerator/plugins/BuildFile.xml are not used/included.

was the list derived in some older release where these packages were still used or is there some longer-range dependency addition?

File: RecoTracker/TkSeedGenerator/plugins/BuildFile.xml (hunk 2 above)
Comment (davidlange6) [reply RecoTracker/TkSeedGenerator/plugins/BuildFile.xml:40]: I used an IB of ~5 days ago. I'll look at what went right or wrong...

File: RecoTracker/TkSeedGenerator/plugins/BuildFile.xml (hunk 3 above)
Comment (davidlange6) [reply RecoTracker/TkSeedGenerator/plugins/BuildFile.xml:15]: in SeedingOTEDProducer.cc

File: RecoTracker/TkSeedGenerator/plugins/BuildFile.xml (hunk 2 above)
Comment (davidlange6) [reply RecoTracker/TkSeedGenerator/plugins/BuildFile.xml:40]: used in SeedingOTEDProducer.cc (and PatternTools is also used in SeedCombiner)

Comment (slava77) [reply RecoTracker/TkSeedGenerator/plugins/BuildFile.xml:40]: > apparently not used in CMSSW_11_1_X_2020-10-27-1100

ehm, why was I looking at an old release cycle

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/31932	c1f32cc 
- technical;

 the updates are apparently all used, sorry for the noise earlier, it came from using an older release cycle for a check

**Comment by qliphy** [comment]
+1

**Pull request merged**